core/src/main/java/org/elasticsearch/rest/action/support/RestTable.java

```diff
@@ -35,19 +35,29 @@
 import java.io.IOException;
 import java.util.*;
 
+import static org.elasticsearch.common.Strings.isEmpty;
+
 /**
  */
 public class RestTable {
 
     public static RestResponse buildResponse(Table table, RestChannel channel) throws Exception {
         RestRequest request = channel.request();
-        XContentType xContentType = XContentType.fromRestContentType(request.param("format", request.header("Content-Type")));
+        XContentType xContentType = XContentType.fromRestContentType(request.param("format", getMediaType(request)));
         if (xContentType != null) {
             return buildXContentBuilder(table, channel);
         }
         return buildTextPlainResponse(table, channel);
     }
 
+    private static String getMediaType(RestRequest request) {
+        if (!isEmpty(request.header("Accept"))) {
+            return request.header("Accept");
+        } else {
+            return request.header("Content-Type");
```

Comment:
With this change, I think that we should completely stop allowing request `Content-Type` to play a role in determining the response `Content-Type`. It was a misuse of the request `Content-Type` and I think that we should take this opportunity to correct it, and add this to the list of breaking changes in the [migration document](https://github.com/elastic/elasticsearch/blob/master/docs/reference/migration/migrate_3_0.asciidoc).